src/Application/Commands/WriteSchemaFile.php

```diff
@@ -0,0 +1,141 @@
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * A file with the full schema kept it git may make it easier for us to see changes etc.
+ */
+#[AsCommand(
+    name: 'matchbot:write-schema-files',
+    description: "Writes a schema file for reference. Inspired by Ruby On Rails's schema.rb file",
+)]
+class WriteSchemaFile extends Command
+{
+    public function __construct(private EntityManagerInterface $em)
+    {
+        parent::__construct(null);
+    }
+
+    public function configure(): void
+    {
+        $this->addOption(
+            'check',
+            description: 'Checks if existing schema files match database, instead of writing new files',
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $hasError = false;
+        $connection = $this->em->getConnection();
+
+        /** @var list<string> $tables */
+        $tables = $connection->executeQuery('SHOW TABLES')->fetchFirstColumn();
+
+        $schemaDir = dirname(__DIR__, 3) . '/schema';
+
+        $fileNames = scandir($schemaDir);
+        \assert(is_array($fileNames));
+
+        $fileNamesOnDisk = array_filter($fileNames, fn(string $fileName) => str_ends_with($fileName, '.sql'));
+
+        $checkModeOn = $input->getOption('check');
+        \assert(is_bool($checkModeOn));
+
+        if ($checkModeOn) {
+            foreach ($fileNamesOnDisk as $fileName) {
+                $tableName = substr($fileName, 0, -4);
+                if (!in_array($tableName, $tables)) {
+                    $error = sprintf(
+                        "%s/%s is on disc but table %s is not in database, please delete or add to db",
+                        $schemaDir,
+                        $fileName,
+                        $tableName
+                    );
+                    $io->error($error);
+                    $hasError = true;
+                }
+            }
+        }
+
+        foreach ($tables as $table) {
+            if (!in_array("$table.sql", $fileNamesOnDisk) && $checkModeOn) {
+                $error = sprintf("Did not find expected file %s/%s.sql", $schemaDir, $table);
+                $io->error($error);
+                $hasError = true;
+            }
+
+            $description = $connection->executeQuery("SHOW CREATE TABLE $table")->fetchAllAssociative();
+            $createTableStatement = $description[0]['Create Table'];
+            \assert(is_string($createTableStatement));
+
+            /* AUTO_INCREMENT varies depending on data (not metadata) in db, not suitable to commit) */
+            $createTableStatement = (string)\preg_replace('/AUTO_INCREMENT=\d+ /', '', $createTableStatement);
+
+            /* Charset and collation vary between local docker setup and circleci. Leave out of files for now to allow
```

Comment:
Presumably we could make these match and save a bit of complexity by extending `cimg/mysql` to set params [as for RDS](https://github.com/thebiggive/infrastructure/blob/a0b30387532ddbe24ffc2ccb2a311f3155957a21/terraform/rds-matchbot.tf#L109-L149), but I think this is also fine.

Reply:
Yes. Having our own docker image just to make this script simpler seems like a bad trade-off, but there probably will be other reasons that we'd want to use the same collation and charset in circleci as in prod.